xpass/xpass.cc

```diff
@@ -221,11 +257,32 @@ void XPassAgent::recv_credit_stop(Packet *pkt) {
 }
 
 void XPassAgent::handle_retransmit() {
+  if (nack_len_ > 0) {
+    send(construct_nack(recv_next_, nack_len_), 0);
+    retransmit_timer_.resched(retransmit_timeout_);
+    return;
+  }
```

Comment:
In this case, we don't need nack_len_, that is, NACK packet do not have to include length information. If sender gets NACK, it only has to send the data as much as possible starting from NACK sequence number.

Reply:
And why don't we add another credit_recv_state_ called XPASS_RECV_RETRANSMIT instead of checking nack_len_?